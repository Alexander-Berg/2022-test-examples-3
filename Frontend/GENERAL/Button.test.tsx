import React from 'react';
import renderer from 'react-test-renderer';

import {
    cnButton,
    ButtonDesktop,
    ButtonTouchPhone,
} from '.';

describe('Button', () => {
    test('cnButton', () => {
        expect(cnButton()).toBe('YpcButton');
    });

    for (const [platform, Button] of Object.entries({
        desktop: ButtonDesktop,
        'touch-phone': ButtonTouchPhone,
    })) {
        describe(platform, () => {
            test('className', () => {
                const tree = renderer
                    .create(
                        <Button className="Test">
                            Text
                        </Button>
                    )
                    .toJSON();

                expect(tree).toMatchSnapshot();
            });

            test('disabled', () => {
                const tree = renderer
                    .create(
                        <Button disabled>
                            <span>Left</span>
                            <span>Right</span>
                        </Button>
                    )
                    .toJSON();

                expect(tree).toMatchSnapshot();
            });
        });
    }
});
