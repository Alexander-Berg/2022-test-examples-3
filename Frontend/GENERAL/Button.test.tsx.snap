// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button desktop className 1`] = `
<button
  className="YpcButton Test"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onPointerEnter={[Function]}
  onPointerLeave={[Function]}
  onTouchStart={[Function]}
  type="button"
>
  Text
</button>
`;

exports[`Button desktop disabled 1`] = `
<button
  className="YpcButton YpcButton_disabled"
  disabled={true}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onPointerEnter={[Function]}
  onPointerLeave={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  <span>
    Left
  </span>
  <span>
    Right
  </span>
</button>
`;

exports[`Button touch-phone className 1`] = `
<button
  className="YpcButton Test"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onPointerEnter={[Function]}
  onPointerLeave={[Function]}
  onTouchStart={[Function]}
  type="button"
>
  Text
</button>
`;

exports[`Button touch-phone disabled 1`] = `
<button
  className="YpcButton YpcButton_disabled"
  disabled={true}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onPointerEnter={[Function]}
  onPointerLeave={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  <span>
    Left
  </span>
  <span>
    Right
  </span>
</button>
`;
